Adding pybind11 bindings for geometry conversions

adityapande-1995 opened the pull request:
This PR aims to add pybinds for ``geometry_conversions.hh``. 

Keeping this a draft until the following checklist is done :

EricCousineau-TRI reviewed (commented):
+@sloretz-tri for initial pass!

> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-:-NNbL89E6z_MQju_ehn4:b584d80)* status: 0 of 4 files reviewed, all discussions resolved (waiting on @sloretz-tri)



<!-- Sent from Reviewable.io -->

sloretz reviewed (changes requested):
> Reviewed 2 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215)* status: 2 of 4 files reviewed, 6 unresolved discussions (waiting on @adityapande-1995 and @sloretz-tri)

___
*[`drake_ros_core/CMakeLists.txt` line 163 at r1](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NNcyVSnDmzPlwxxTQvv:-NNcyVSnDmzPlwxxTQvw:b-43hmsf) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/630a16b3474e0490922577a98eb67a4309ed6a9e/drake_ros_core/CMakeLists.txt#L163)):*
> ```CMake
>   target_link_libraries(test_geometry_conversions drake_ros_core)
> 
>   ament_add_pytest_test(test_geometry_conversions_py
> ```

Mind adding this to the `BUILD.bazel` as a `ros_py_test`?
___
*[`drake_ros_core/include/drake_ros_core/geometry_typecast.h` line 102 at r1](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NNd2PPv80Biiiw9CKeR:-NNd2PPv80Biiiw9CKeS:b-ea8l8t) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/630a16b3474e0490922577a98eb67a4309ed6a9e/drake_ros_core/include/drake_ros_core/geometry_typecast.h#L102)):*
> ```objc
>     public:
>         PYBIND11_TYPE_CASTER(Eigen::Vector3d,
>             _("numpy.array"));
> ```

I think this conversion must exist already, maybe via https://pybind11.readthedocs.io/en/stable/advanced/cast/eigen.html


I see it's already being used (via `Eigen::Ref`) in Drake: https://github.com/RobotLocomotion/drake/blob/7251918f5ba66176e52f9215e8c5ecba55a7a8ec/bindings/pydrake/multibody/inverse_kinematics_py.cc#L342

I tested removing it and I found it works if I include `<pybind11/eigen.h>`, though the returned arrays are a little different. `array([1, 2, 3])` instead of `array([[1], [2], [3]])` so the test expectations need to be updated.
___
*[`drake_ros_core/include/drake_ros_core/geometry_typecast.h` line 323 at r1](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NNdAw-w4xRPQqU8Leyq:-NNdAw-w4xRPQqU8Leyr:b2znl0e) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/630a16b3474e0490922577a98eb67a4309ed6a9e/drake_ros_core/include/drake_ros_core/geometry_typecast.h#L323)):*
> ```objc
>     // Typecasting between numpy.array (Python) and
>     // drake::Vector6d (C++)
>     template <> struct type_caster<drake::Vector6d> {
> ```

Same comment here about not needing this one. `drake::Vector6d` is also just an alias for an Eigen type. I found everything still runs when I comment this out, but the `array()` shape is again different so test expectations will need to be updated.
___
*[`drake_ros_core/test/test_geometry_conversions_py.py` line 85 at r1](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NNd88So1nGxn8wJP6Km:-NNd88So1nGxn8wJP6Kn:b77jbwz) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/630a16b3474e0490922577a98eb67a4309ed6a9e/drake_ros_core/test/test_geometry_conversions_py.py#L85)):*
> ```Python
>     quaternion_expected = \
>             _drake_ros_core.rotation_matrix_to_ros_quaternion(rot_matrix_converted)
>     assert (quaternion_expected == q)
> ```

nit, no need for parenthesis. `assert` is a statement rather than a function in Python.
___
*[`drake_ros_core/test/test_geometry_conversions_py.py` line 88 at r1](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NNd7hRN4H4lhlytYKy9:-NNd7hRN4H4lhlytYKyA:b-n4fgfz) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/630a16b3474e0490922577a98eb67a4309ed6a9e/drake_ros_core/test/test_geometry_conversions_py.py#L88)):*
> ```Python
> 
> def test_pose():
>     # ROS pose to rigid transform.
> ```

for Python tests I'd recommend splitting them into individual functions so that we get more information when a test fails. For example, right now if the expectation on `ros_pose_to_rigid_transform` fails then we get no information about `rigid_transform_to_ros_pose`.
___
*[`drake_ros_core/test/test_geometry_conversions_py.py` line 170 at r1](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NNd5v_23cvraOYyzJ8x:-NNd5v_23cvraOYyzJ8y:bjaont8) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/630a16b3474e0490922577a98eb67a4309ed6a9e/drake_ros_core/test/test_geometry_conversions_py.py#L170)):*
> ```Python
>                                  [0.33333333, 0.93333333, 0.13333333]]))
> 
>     ros_pose_expected = _drake_ros_core.isometry3_to_ros_pose(isometry3)
> ```

I get a runtime error here. The "Measure of orthonormality error" is so small I wonder if it's a rounding issue. Do you see this on your machine too?

```
E       RuntimeError: Error: Rotation matrix is not orthonormal.
E         Measure of orthonormality error: 4.6666667641304116e-08  (near-zero is good).
E         To calculate the proper orthonormal rotation matrix closest to the alleged rotation matrix, use the SVD (expensive) static method RotationMatrix<T>::ProjectToRotationMatrix(), or for a less expensive (but not necessarily closest) rotation matrix, use RotationMatrix<T>(RotationMatrix<T>::ToQuaternion<T>(your_matrix)). Alternatively, if using quaternions, ensure the quaternion is normalized.

test/test_geometry_conversions_py.py:171: RuntimeError
```


<!-- Sent from Reviewable.io -->

adityapande-1995 commented:
## TODO
- [ ] Run the python test using bazel, add to ``BUILD.bazel``
- [ ] Remove Eigen to numpy array line, account for changed array shape.
- [ ] ``assert`` paranthesis remove
- [ ] Split tests into individual ones
- [ ] Isometry 3 to ros pose orthonormal error

EricCousineau-TRI reviewed (changes requested):
First pass done on my side as well!

> Reviewed 2 of 4 files at r1, 3 of 3 files at r2.
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-:-NOPeVrQ5oC3Py20Vxt4:brufglz)* status: all files reviewed (commit messages unreviewed), 13 unresolved discussions (waiting on @adityapande-1995, @sloretz, and @sloretz-tri)

___
*[`drake_ros_core/CMakeLists.txt` line 163 at r1](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NNcyVSnDmzPlwxxTQvv:-NOPeQTl7PC38idWAYHP:b9zlnfj) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/630a16b3474e0490922577a98eb67a4309ed6a9e/drake_ros_core/CMakeLists.txt#L163)):*
<details><summary><i>Previously, sloretz (Shane Loretz) wrote…</i></summary><blockquote>

Mind adding this to the `BUILD.bazel` as a `ros_py_test`?
</blockquote></details>

I think this is done?
___
*[`drake_ros_core/include/drake_ros_core/geometry_typecast.h` line 13 at r2](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NOPcaB08I-0cVF8kVc0:-NOPcaB08I-0cVF8kVc1:b-h1i21b) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/a01773ef17edaecc3c36c703e1041450279135d8/drake_ros_core/include/drake_ros_core/geometry_typecast.h#L13)):*
> ```objc
> namespace py = pybind11;
> 
> namespace PYBIND11_NAMESPACE {
> ```

Could you try the automated route through serialization, rather than manually re-enumerating fields?
See my draft for rough example: https://github.com/RobotLocomotion/drake-ros/pull/169

Alternatively, rather than using serialization outright, if there's some way to use the IDL introspection mechanisms, maybe that's better? (e.g. getting field name, type, and offset?)

Or, just to move forward, can you add a TODO to that effect?
___
*[`drake_ros_core/src/module_py.cc` line 95 at r2](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NOPdMqKBoTxF--3Hfil:-NOPdMqKBoTxF--3Hfim:b1j5kk8) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/a01773ef17edaecc3c36c703e1041450279135d8/drake_ros_core/src/module_py.cc#L95)):*
> ```cpp
>   py::module::import("pydrake.math");
>   py::module::import("pydrake.common.eigen_geometry");
>   py::module::import("numpy");
> ```

nit Would you be able to maintain import order of "builtin, third party, second(ish) party, first party"?
i.e., can you place `numpy` before `pydrake`?
___
*[`drake_ros_core/src/module_py.cc` line 175 at r2](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NOPd13B2L4ssEwRtGTA:-NOPd13C7-AOBEm8nnCq:b26mkh8) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/a01773ef17edaecc3c36c703e1041450279135d8/drake_ros_core/src/module_py.cc#L175)):*
> ```cpp
>   // Python bindings for geometry conversions.
>   // Vector / Translation functions.
>   m.def("ros_point_to_vector3", &drake_ros_core::RosPointToVector3);
> ```

While this is great for Python PEP 8, we actually don't (yet?) use different names between C++ and Python in Drake:
https://drake.mit.edu/python_bindings.html

Can you change these to be the same casing as in C++?
___
*[`drake_ros_core/test/test_geometry_conversions_py.py` line 5 at r2](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NOPdf8W9QeyH7jHcMJN:-NOPdf8X6JPblpfsZmRC:b-lxnc4w) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/a01773ef17edaecc3c36c703e1041450279135d8/drake_ros_core/test/test_geometry_conversions_py.py#L5)):*
> ```Python
> import pydrake.multibody.math
> import pydrake.common.eigen_geometry
> import _drake_ros_core
> ```

Can you use public spelling of the module for this?
___
*[`drake_ros_core/test/test_geometry_conversions_py.py` line 6 at r2](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NOPdgmF1sNxIEuJ2Uzw:-NOPdgmF1sNxIEuJ2Uzx:b71rixy) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/a01773ef17edaecc3c36c703e1041450279135d8/drake_ros_core/test/test_geometry_conversions_py.py#L6)):*
> ```Python
> import pydrake.common.eigen_geometry
> import _drake_ros_core
> import numpy as np
> ```

Can you use import order for "builtin, third party, second(ish) party, first part"? i.e.,
```
()
(numpy, pytest)
(pydrake)
(drake_ros_core)
```
___
*[`drake_ros_core/test/test_geometry_conversions_py.py` line 18 at r2](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NOPe7me5J1BeCGyq4aH:-NOPe7me5J1BeCGyq4aI:b6ihstx) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/a01773ef17edaecc3c36c703e1041450279135d8/drake_ros_core/test/test_geometry_conversions_py.py#L18)):*
> ```Python
>     array_expected = np.array([1.12, 2.34, 3.456])
> 
>     assert (array_expected == array_converted).all()
> ```

Rather than `assert {expr}.all()`, consider using NumPy testing affordances:
https://numpy.org/doc/stable/reference/routines.testing.html
___
*[`drake_ros_core/test/test_geometry_conversions_py.py` line 25 at r2](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NOPdwIH9Nog_IMr5ejr:-NOPdwIH9Nog_IMr5ejs:b9aqcw3) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/a01773ef17edaecc3c36c703e1041450279135d8/drake_ros_core/test/test_geometry_conversions_py.py#L25)):*
> ```Python
>     p.y = 2.34
>     p.z = 3.456
>     point_expected = \
> ```

nit PEP 8: Line continuation should use parentheses, not forward slash.

https://peps.python.org/pep-0008/#maximum-line-length


<!-- Sent from Reviewable.io -->

adityapande-1995 commented:
## Review checklist : 
- [ ] Change order of ``numpy`` and ``pydrake`` in ``module_py.cc``
- [ ] Change function name casing to C++ style
- [ ] PEP 8 line continuation
- [ ] Numpy testing methods rather than ``.all()``
- [ ] Fix import order in test file
- [ ] Use public spelling of ``_drake_ros_core``

EricCousineau-TRI reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215)* status: 1 of 5 files reviewed, 14 unresolved discussions (waiting on @adityapande-1995, @sloretz, and @sloretz-tri)

*[a discussion](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NOyNn0F-_YQL6bOuTC4:-NOyNn0G7mcySLlS3X_8:b-sxgylh) (no related file):*
The failures on CI make me a bit suspicious - would you up for rebase-squashing your commit onto latest `main`?

If you wish to access old revisions of your commit tree, note that Reviewable has the ability to fetch old revisions:
https://docs.reviewable.io/tips.html#easy-local-revision-checkout

If you also didn't know, there's `git reflog` that can tell you reference history, even across rebases, etc.:
https://git-scm.com/docs/git-reflog

___
*[a discussion](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NOyOd3j-l55Jcz6p3tb:-NOyOd3k-yiiT5GnnUOW:bfvmu05) (no related file):*
FYI @adityapande-1995 It's not immediately clear to me when I should re-engage on the PR.

Can you post, via Reviewable, when you've addressed comments on each specific discussion item? e.g. "Done" or "Working" if you still need to change it?

___


<!-- Sent from Reviewable.io -->

adityapande-1995 commented:
@EricCousineau-TRI yup, will do. I'm still working on the generic typecasting, converting this to a draft for now.

adityapande-1995 commented:
*[`drake_ros_core/include/drake_ros_core/geometry_typecast.h` line 102 at r1](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NNd2PPv80Biiiw9CKeR:-NPUFQ1QCP8p4Gcemm8_:b-896fix) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/630a16b3474e0490922577a98eb67a4309ed6a9e/drake_ros_core/include/drake_ros_core/geometry_typecast.h#L102)):*
<details><summary><i>Previously, sloretz (Shane Loretz) wrote…</i></summary><blockquote>

I think this conversion must exist already, maybe via https://pybind11.readthedocs.io/en/stable/advanced/cast/eigen.html


I see it's already being used (via `Eigen::Ref`) in Drake: https://github.com/RobotLocomotion/drake/blob/7251918f5ba66176e52f9215e8c5ecba55a7a8ec/bindings/pydrake/multibody/inverse_kinematics_py.cc#L342

I tested removing it and I found it works if I include `<pybind11/eigen.h>`, though the returned arrays are a little different. `array([1, 2, 3])` instead of `array([[1], [2], [3]])` so the test expectations need to be updated.
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

adityapande-1995 commented:
*[`drake_ros_core/include/drake_ros_core/geometry_typecast.h` line 323 at r1](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NNdAw-w4xRPQqU8Leyq:-NPUFRt4CyassehE6Zvt:b-896fix) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/630a16b3474e0490922577a98eb67a4309ed6a9e/drake_ros_core/include/drake_ros_core/geometry_typecast.h#L323)):*
<details><summary><i>Previously, sloretz (Shane Loretz) wrote…</i></summary><blockquote>

Same comment here about not needing this one. `drake::Vector6d` is also just an alias for an Eigen type. I found everything still runs when I comment this out, but the `array()` shape is again different so test expectations will need to be updated.
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

adityapande-1995 commented:
*[`drake_ros_core/test/test_geometry_conversions_py.py` line 5 at r2](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NOPdf8W9QeyH7jHcMJN:-NPUFTZt9NC1PLRdfS30:b-896fix) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/a01773ef17edaecc3c36c703e1041450279135d8/drake_ros_core/test/test_geometry_conversions_py.py#L5)):*
<details><summary><i>Previously, EricCousineau-TRI (Eric Cousineau) wrote…</i></summary><blockquote>

Can you use public spelling of the module for this?
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

adityapande-1995 commented:
*[`drake_ros_core/test/test_geometry_conversions_py.py` line 6 at r2](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NOPdgmF1sNxIEuJ2Uzw:-NPUFUIND5b9mp5zdHiD:b-896fix) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/a01773ef17edaecc3c36c703e1041450279135d8/drake_ros_core/test/test_geometry_conversions_py.py#L6)):*
<details><summary><i>Previously, EricCousineau-TRI (Eric Cousineau) wrote…</i></summary><blockquote>

Can you use import order for "builtin, third party, second(ish) party, first part"? i.e.,
```
()
(numpy, pytest)
(pydrake)
(drake_ros_core)
```
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

adityapande-1995 commented:
*[`drake_ros_core/test/test_geometry_conversions_py.py` line 18 at r2](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NOPe7me5J1BeCGyq4aH:-NPUFVh81pfhWoyIxl7h:b-896fix) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/a01773ef17edaecc3c36c703e1041450279135d8/drake_ros_core/test/test_geometry_conversions_py.py#L18)):*
<details><summary><i>Previously, EricCousineau-TRI (Eric Cousineau) wrote…</i></summary><blockquote>

Rather than `assert {expr}.all()`, consider using NumPy testing affordances:
https://numpy.org/doc/stable/reference/routines.testing.html
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

adityapande-1995 commented:
*[`drake_ros_core/test/test_geometry_conversions_py.py` line 88 at r1](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NNd7hRN4H4lhlytYKy9:-NPUFZTF11TbP5d1qz0u:b-896fix) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/630a16b3474e0490922577a98eb67a4309ed6a9e/drake_ros_core/test/test_geometry_conversions_py.py#L88)):*
<details><summary><i>Previously, sloretz (Shane Loretz) wrote…</i></summary><blockquote>

for Python tests I'd recommend splitting them into individual functions so that we get more information when a test fails. For example, right now if the expectation on `ros_pose_to_rigid_transform` fails then we get no information about `rigid_transform_to_ros_pose`.
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

adityapande-1995 commented:
*[`drake_ros_core/test/test_geometry_conversions_py.py` line 170 at r1](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NNd5v_23cvraOYyzJ8x:-NPUFbBs85aD9vOfHKD-:b-896fix) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/630a16b3474e0490922577a98eb67a4309ed6a9e/drake_ros_core/test/test_geometry_conversions_py.py#L170)):*
<details><summary><i>Previously, sloretz (Shane Loretz) wrote…</i></summary><blockquote>

I get a runtime error here. The "Measure of orthonormality error" is so small I wonder if it's a rounding issue. Do you see this on your machine too?

```
E       RuntimeError: Error: Rotation matrix is not orthonormal.
E         Measure of orthonormality error: 4.6666667641304116e-08  (near-zero is good).
E         To calculate the proper orthonormal rotation matrix closest to the alleged rotation matrix, use the SVD (expensive) static method RotationMatrix<T>::ProjectToRotationMatrix(), or for a less expensive (but not necessarily closest) rotation matrix, use RotationMatrix<T>(RotationMatrix<T>::ToQuaternion<T>(your_matrix)). Alternatively, if using quaternions, ensure the quaternion is normalized.

test/test_geometry_conversions_py.py:171: RuntimeError
```
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

adityapande-1995 commented:
*[`drake_ros_core/src/module_py.cc` line 175 at r2](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NOPd13B2L4ssEwRtGTA:-NPUFfV-61SUXPf3hpDl:b-896fix) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/a01773ef17edaecc3c36c703e1041450279135d8/drake_ros_core/src/module_py.cc#L175)):*
<details><summary><i>Previously, EricCousineau-TRI (Eric Cousineau) wrote…</i></summary><blockquote>

While this is great for Python PEP 8, we actually don't (yet?) use different names between C++ and Python in Drake:
https://drake.mit.edu/python_bindings.html

Can you change these to be the same casing as in C++?
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

adityapande-1995 commented:
I'm note sure about the reason for pycodestyle and visual marker errors in CI here. Open for a new round of review @sloretz @EricCousineau-TRI !

adityapande-1995 commented:
*[`drake_ros/core/CMakeLists.txt` line 14 at r4](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NPZs3eBC_gsCp5QLNLW:-NPd0Wph5ahlHHAMex6Y:b-896fix) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/f62c7c50ca6cc849d218bf61fde811be7df011a4/drake_ros/core/CMakeLists.txt#L14)):*
<details><summary><i>Previously, EricCousineau-TRI (Eric Cousineau) wrote…</i></summary><blockquote>

This is for `pybind`, and its name should reflect that.

See "Target Conventions"
https://drake.mit.edu/doxygen_cxx/group__python__bindings.html

Can you rename to `geometry_conversions_pybind.h`?
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

sloretz reviewed (commented):
> Reviewed 7 of 8 files at r4, 3 of 3 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215)* status: all files reviewed, 21 unresolved discussions (waiting on @adityapande-1995, @EricCousineau-TRI, and @sloretz-tri)

___
*[`drake_ros/core/BUILD.bazel` line 136 at r4](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NPZbw8o2mXG4jtk8Vja:-NPZbw8o2mXG4jtk8Vjb:b-oxhtdn) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/f62c7c50ca6cc849d218bf61fde811be7df011a4/drake_ros/core/BUILD.bazel#L136)):*
> ```Starlark
>         "@drake//bindings/pydrake",
>         "@ros2//:geometry_msgs_py",
>         "@ros2//:rclpy_py",
> ```

nit, I think this might be unused too.
___
*[`drake_ros/core/BUILD.bazel` line 137 at r4](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NPZbl-c-Dor9HaibOCV:-NPZbl-c-Dor9HaibOCW:b-df3d03) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/f62c7c50ca6cc849d218bf61fde811be7df011a4/drake_ros/core/BUILD.bazel#L137)):*
> ```Starlark
>         "@ros2//:geometry_msgs_py",
>         "@ros2//:rclpy_py",
>         "@ros2//:test_msgs_py",
> ```

nit, I think this dependency is unused - could remove it.
___
*[`drake_ros_core/CMakeLists.txt` line 163 at r1](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NNcyVSnDmzPlwxxTQvv:-NPdicAE43XYtjrZ-eov:b-bwpp4l) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/630a16b3474e0490922577a98eb67a4309ed6a9e/drake_ros_core/CMakeLists.txt#L163)):*
<details><summary><i>Previously, EricCousineau-TRI (Eric Cousineau) wrote…</i></summary><blockquote>

I think this is done?
</blockquote></details>

I think it's in the `BUILD.bazel` now, but I no longer see it in the `CMakeLists.txt`. @EricCousineau-TRI think it needs to be tested in both, or is bazel testing enough?
___
*[`drake_ros/core/test/test_geometry_conversions_py.py` line 30 at r5](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NPdmxv_3wKe4twxlstx:-NPdmxv_3wKe4twxlsty:b744q7o) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/5fd3f1eebe33c45e112ed401ce6939a51f93fbef/drake_ros/core/test/test_geometry_conversions_py.py#L30)):*
> ```Python
>     p.y = 2.34
>     p.z = 3.456
>     point_expected = drake_ros_core.Vector3ToRosPoint(
> ```

nit, this is the converted point. The expected point is `p`.
___
*[`drake_ros/core/test/test_geometry_conversions_py.py` line 50 at r5](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NPdnDdy6EMNwSYnW8yy:-NPdnDdy6EMNwSYnW8yz:by98bqg) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/5fd3f1eebe33c45e112ed401ce6939a51f93fbef/drake_ros/core/test/test_geometry_conversions_py.py#L50)):*
> ```Python
>     v.y = 2.50
>     v.z = 3.75
>     ros_vec3_expected = drake_ros_core.Vector3ToRosVector3(
> ```

nit, this is the converted value, not the expected one.
___
*[`drake_ros/core/test/test_geometry_conversions_py.py` line 61 at r5](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NPdnNz39kxBZ94X3k8t:-NPdnNz39kxBZ94X3k8u:b-brzaty) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/5fd3f1eebe33c45e112ed401ce6939a51f93fbef/drake_ros/core/test/test_geometry_conversions_py.py#L61)):*
> ```Python
>     q.z = 0.3
>     q.w = 0.4
>     pydrake_quaternon_expected = drake_ros_core.RosQuaternionToQuaternion(q)
> ```

nit, this is the converted quaternion. Also, `quaternon` -> `quaternion`.
___
*[`drake_ros/core/test/test_geometry_conversions_py.py` line 102 at r5](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NPdnf569YOt1rzUqgfV:-NPdnf572PBtWcJxf-uE:b-1ngp75) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/5fd3f1eebe33c45e112ed401ce6939a51f93fbef/drake_ros/core/test/test_geometry_conversions_py.py#L102)):*
> ```Python
>                   [1.0, 0.0, 0.0],
>                   [0.0, 1.0, 0.0]]))
>     quaternion_expected = drake_ros_core.RotationMatrixToRosQuaternion(
> ```

nit, this is the converted, not the expected quaternion
___
*[`drake_ros/core/test/test_geometry_conversions_py.py` line 116 at r5](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NPdnlr46c-Mbh9Y4PxX:-NPdnlr46c-Mbh9Y4PxY:by98bqg) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/5fd3f1eebe33c45e112ed401ce6939a51f93fbef/drake_ros/core/test/test_geometry_conversions_py.py#L116)):*
> ```Python
>     ros_pose.orientation.y = 3.0
>     ros_pose.orientation.z = 4.0
>     rigid_transform_expected = drake_ros_core.RosPoseToRigidTransform(ros_pose)
> ```

nit, this is the converted value, not the expected one.
___
*[`drake_ros/core/test/test_geometry_conversions_py.py` line 133 at r5](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NPdnyKJAXS-trzsuzzG:-NPdnyKJAXS-trzsuzzH:b1ud7m1) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/5fd3f1eebe33c45e112ed401ce6939a51f93fbef/drake_ros/core/test/test_geometry_conversions_py.py#L133)):*
> ```Python
>                  3.0/np.sqrt(30), 4.0/np.sqrt(30)]))
> 
>     ros_pose_expected = drake_ros_core.RigidTransformToRosPose(rigid_transform)
> ```

nit, converted not expected. Or just `ros_pose` would be fine here IMO.
___
*[`drake_ros/core/test/test_geometry_conversions_py.py` line 152 at r5](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NPdo3rjDG2GVAmQrzzi:-NPdo3rjDG2GVAmQrzzj:b-pox4yz) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/5fd3f1eebe33c45e112ed401ce6939a51f93fbef/drake_ros/core/test/test_geometry_conversions_py.py#L152)):*
> ```Python
>     ros_transform.rotation.y = 3.0
>     ros_transform.rotation.z = 4.0
>     rigid_transform_expected = drake_ros_core.RosTransformToRigidTransform(
> ```

nit, same comment about this being converted
___
*[`drake_ros/core/test/test_geometry_conversions_py.py` line 168 at r5](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NPdo6HBCkMkiNsEodJ_:-NPdo6HBCkMkiNsEodJa:b-f4aocn) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/5fd3f1eebe33c45e112ed401ce6939a51f93fbef/drake_ros/core/test/test_geometry_conversions_py.py#L168)):*
> ```Python
>                 [1.0/np.sqrt(30), 2.0/np.sqrt(30),
>                  3.0/np.sqrt(30), 4.0/np.sqrt(30)]))
>     ros_transform_expected = drake_ros_core.RigidTransformToRosTransform(
> ```

nit, same comment about this being converted.
___
*[`drake_ros/core/test/test_geometry_conversions_py.py` line 188 at r5](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NPdo8w-2mASXTz1rQoV:-NPdo8w-2mASXTz1rQoW:b-pox4yz) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/5fd3f1eebe33c45e112ed401ce6939a51f93fbef/drake_ros/core/test/test_geometry_conversions_py.py#L188)):*
> ```Python
>     ros_pose.orientation.y = 3.0
>     ros_pose.orientation.z = 4.0
>     isometry_expected = drake_ros_core.RosPoseToIsometry3(ros_pose)
> ```

nit, same comment about this being converted
___
*[`drake_ros/core/test/test_geometry_conversions_py.py` line 223 at r5](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NPdoIOE7bJvRDEKgAGf:-NPdoIOE7bJvRDEKgAGg:b-pox4yz) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/5fd3f1eebe33c45e112ed401ce6939a51f93fbef/drake_ros/core/test/test_geometry_conversions_py.py#L223)):*
> ```Python
>     ros_transform.rotation.y = 3.0
>     ros_transform.rotation.z = 4.0
>     isometry_expected = drake_ros_core.RosTransformToIsometry3(ros_transform)
> ```

nit, same comment about this being converted
___
*[`drake_ros/core/test/test_geometry_conversions_py.py` line 238 at r5](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NPdoKxm8zUD73FW5TNg:-NPdoKxm8zUD73FW5TNh:b-f4aocn) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/5fd3f1eebe33c45e112ed401ce6939a51f93fbef/drake_ros/core/test/test_geometry_conversions_py.py#L238)):*
> ```Python
>                 [1/np.sqrt(30), 2/np.sqrt(30), 3/np.sqrt(30), 4/np.sqrt(30)]))
> 
>     ros_transform_expected = drake_ros_core.Isometry3ToRosTransform(isometry3)
> ```

nit, same comment about this being converted.
___
*[`drake_ros/core/test/test_geometry_conversions_py.py` line 288 at r5](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NPdoROf7cTHGBCKH5sc:-NPdoROf7cTHGBCKH5sd:b-f4aocn) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/5fd3f1eebe33c45e112ed401ce6939a51f93fbef/drake_ros/core/test/test_geometry_conversions_py.py#L288)):*
> ```Python
>     t.angular.y = 22.22
>     t.angular.z = 33.33
>     vec6_expected = drake_ros_core.RosTwistToVector6(t)
> ```

nit, same comment about this being converted.
___
*[`drake_ros/core/test/test_geometry_conversions_py.py` line 302 at r5](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NPdoV1Y1szaHi3lE9fH:-NPdoV1Y1szaHi3lE9fI:b-pox4yz) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/5fd3f1eebe33c45e112ed401ce6939a51f93fbef/drake_ros/core/test/test_geometry_conversions_py.py#L302)):*
> ```Python
>     t.angular.y = 22.22
>     t.angular.z = 33.33
>     ros_twist_expected = drake_ros_core.Vector6ToRosTwist(
> ```

nit, same comment about this being converted
___
*[`drake_ros/core/test/test_geometry_conversions_py.py` line 315 at r5](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NPdoYeg1UvE3wlcM3WH:-NPdoYeg1UvE3wlcM3WI:b-pox4yz) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/5fd3f1eebe33c45e112ed401ce6939a51f93fbef/drake_ros/core/test/test_geometry_conversions_py.py#L315)):*
> ```Python
>     a.angular.y = 22.22
>     a.angular.z = 33.33
>     vec6_expected = drake_ros_core.RosAccelToVector6(a)
> ```

nit, same comment about this being converted
___
*[`drake_ros/core/test/test_geometry_conversions_py.py` line 328 at r5](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NPdob3K8EtKy3tyzR7U:-NPdob3K8EtKy3tyzR7V:b-f4aocn) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/5fd3f1eebe33c45e112ed401ce6939a51f93fbef/drake_ros/core/test/test_geometry_conversions_py.py#L328)):*
> ```Python
>     a.angular.y = 22.22
>     a.angular.z = 33.33
>     ros_accel_expected = drake_ros_core.Vector6ToRosAccel(
> ```

nit, same comment about this being converted.
___
*[`drake_ros/core/test/test_geometry_conversions_py.py` line 373 at r5](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NPdoekHAt4NMq8ilLX-:-NPdoekHAt4NMq8ilLX0:b-pox4yz) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/5fd3f1eebe33c45e112ed401ce6939a51f93fbef/drake_ros/core/test/test_geometry_conversions_py.py#L373)):*
> ```Python
>     w.torque.y = 22.22
>     w.torque.z = 33.33
>     vec6_expected = drake_ros_core.RosWrenchToVector6(w)
> ```

nit, same comment about this being converted
___
*[`drake_ros/core/test/test_geometry_conversions_py.py` line 386 at r5](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NPdoh3I-xhdHgPlhdO1:-NPdoh3I-xhdHgPlhdO2:b-f4aocn) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/5fd3f1eebe33c45e112ed401ce6939a51f93fbef/drake_ros/core/test/test_geometry_conversions_py.py#L386)):*
> ```Python
>     w.torque.y = 22.22
>     w.torque.z = 33.33
>     ros_wrench_expected = drake_ros_core.Vector6ToRosWrench(
> ```

nit, same comment about this being converted.
___
*[`drake_ros_core/test/test_geometry_conversions_py.py` line 170 at r1](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NNd5v_23cvraOYyzJ8x:-NPdoEPP44U64RFFCHEI:b-pox4yz) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/630a16b3474e0490922577a98eb67a4309ed6a9e/drake_ros_core/test/test_geometry_conversions_py.py#L170)):*
<details><summary><i>Previously, adityapande-1995 (Aditya Pande) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

nit, same comment about this being converted


<!-- Sent from Reviewable.io -->

adityapande-1995 commented:
*[`drake_ros/core/BUILD.bazel` line 137 at r4](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NPZbl-c-Dor9HaibOCV:-NPp6YW9DVrZfYpeW6LW:bb6un1j) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/f62c7c50ca6cc849d218bf61fde811be7df011a4/drake_ros/core/BUILD.bazel#L137)):*
<details><summary><i>Previously, sloretz (Shane Loretz) wrote…</i></summary><blockquote>

nit, I think this dependency is unused - could remove it.
</blockquote></details>

Done


<!-- Sent from Reviewable.io -->

adityapande-1995 commented:
*[`drake_ros/core/BUILD.bazel` line 136 at r4](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NPZbw8o2mXG4jtk8Vja:-NPp6O6bBitEgKrrMMWU:b-x689xs) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/f62c7c50ca6cc849d218bf61fde811be7df011a4/drake_ros/core/BUILD.bazel#L136)):*
<details><summary><i>Previously, sloretz (Shane Loretz) wrote…</i></summary><blockquote>

nit, I think this might be unused too.
</blockquote></details>

This is a an indirect dependency of drake_ros_core, and the test fails without this.


<!-- Sent from Reviewable.io -->

adityapande-1995 commented:
*[`drake_ros_core/CMakeLists.txt` line 163 at r1](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NNcyVSnDmzPlwxxTQvv:-NPp7-E--jpmez39-x7Z:bkpb00d) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/630a16b3474e0490922577a98eb67a4309ed6a9e/drake_ros_core/CMakeLists.txt#L163)):*
<details><summary><i>Previously, sloretz (Shane Loretz) wrote…</i></summary><blockquote>

I think it's in the `BUILD.bazel` now, but I no longer see it in the `CMakeLists.txt`. @EricCousineau-TRI think it needs to be tested in both, or is bazel testing enough?
</blockquote></details>

Added the ament_pytest as well : https://github.com/RobotLocomotion/drake-ros/pull/215/commits/2988101d56ad580d7ff0858acd2e1eae7a938da1


<!-- Sent from Reviewable.io -->

adityapande-1995 commented:
*[`drake_ros/core/test/test_geometry_conversions_py.py` line 61 at r5](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NPdnNz39kxBZ94X3k8t:-NPpARhk6I6O4qI6nra2:b-oga9xt) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/5fd3f1eebe33c45e112ed401ce6939a51f93fbef/drake_ros/core/test/test_geometry_conversions_py.py#L61)):*
<details><summary><i>Previously, sloretz (Shane Loretz) wrote…</i></summary><blockquote>

nit, this is the converted quaternion. Also, `quaternon` -> `quaternion`.
</blockquote></details>

Corrected the typo


<!-- Sent from Reviewable.io -->

adityapande-1995 commented:
*[`drake_ros_core/test/test_geometry_conversions_py.py` line 170 at r1](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NNd5v_23cvraOYyzJ8x:-NPpFOeu887Axb44-VSR:b-896fjb) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/630a16b3474e0490922577a98eb67a4309ed6a9e/drake_ros_core/test/test_geometry_conversions_py.py#L170)):*
<details><summary><i>Previously, sloretz (Shane Loretz) wrote…</i></summary><blockquote>

nit, same comment about this being converted
</blockquote></details>

Done 


<!-- Sent from Reviewable.io -->

adityapande-1995 commented:
Working on the test failures here, converting to a draft.

adityapande-1995 commented:
I have 2 weird failures here : One about pycodestyle and other about visual markers, not sure if those are related to changes made here.

adityapande-1995 commented:
*[`drake_ros/core/BUILD.bazel` line 136 at r4](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NPZbw8o2mXG4jtk8Vja:-NPsQqc31NEVLnqMqtvv:bjyjghy) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/f62c7c50ca6cc849d218bf61fde811be7df011a4/drake_ros/core/BUILD.bazel#L136)):*
<details><summary><i>Previously, adityapande-1995 (Aditya Pande) wrote…</i></summary><blockquote>

This is a an indirect dependency of drake_ros_core, and the test fails without this.
</blockquote></details>

Marking this as resolved after the discussion with Eric.


<!-- Sent from Reviewable.io -->

sloretz reviewed (commented):
The pycodestyle error should be fixed by: https://github.com/ament/ament_lint/pull/435

It will take a bit of time for that bug to be backported to Humble. In the meantime, maybe it makes sense to exclude this file with a `TODO` saying to reenable it when ament/ament_lint#435 is released into Humble.

https://github.com/RobotLocomotion/drake-ros/blob/94a32508bb7e1e1cfb5c422ac7a64bc836d9dd3d/drake_ros/CMakeLists.txt#L33

I think changing that to `ament_pycodestyle(--config pycodestyle.ini --exclude "${CMAKE_CURRENT_SOURCE_DIR}/core/test")` might do it

> Reviewed 1 of 1 files at r7, 3 of 3 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-:-NPsyWDhCSXQdMsi3Pjq:b-aperso)* status: all files reviewed, 3 unresolved discussions (waiting on @adityapande-1995, @EricCousineau-TRI, and @sloretz-tri)



<!-- Sent from Reviewable.io -->

EricCousineau-TRI reviewed (commented):
> other about visual markers

That should be resolved once #242 lands.

> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-:-NPtEY_z6a7AEc3uNE3q:b-tep70k)* status: all files reviewed, 3 unresolved discussions (waiting on @adityapande-1995 and @sloretz-tri)



<!-- Sent from Reviewable.io -->

EricCousineau-TRI reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215)* status: all files reviewed, 4 unresolved discussions (waiting on @adityapande-1995 and @sloretz-tri)

*[a discussion](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NPtgjeXA4grqntWOAtG:-NPtgjeXA4grqntWOAtH:b-lszval) (no related file):*
@adityapande-1995 Would you be OK with waiting until #238 lands, then rebasing this PR on top of `main`?

___
___
*[`drake_ros/core/test/test_geometry_conversions_py.py` line 423 at r10](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NPth-qM2jS9yXT1t5rT:-NPth-qNAo6AmtVJJslS:bssq15k) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/9b9d79cc924df617cf04b27f85957e6da9f58aee/drake_ros/core/test/test_geometry_conversions_py.py#L423)):*
> ```Python
>                 )
>             )
>     assert ros_wrench_converted == w
> ```

Per #243, we should ensure this has a clause for Bazel to know to execute this code. From #238, it should be something like:
```
if __name__ == '__main__':
    sys.exit(pytest.main(sys.argv))
```


<!-- Sent from Reviewable.io -->

adityapande-1995 commented:
<details><summary><i>Previously, EricCousineau-TRI (Eric Cousineau) wrote…</i></summary><blockquote>

@adityapande-1995 Would you be OK with waiting until #238 lands, then rebasing this PR on top of `main`?
</blockquote></details>

Done, rebased and moved the python test.




<!-- Sent from Reviewable.io -->

adityapande-1995 commented:
*[`drake_ros/core/test/test_geometry_conversions_py.py` line 423 at r10](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NPth-qM2jS9yXT1t5rT:-NPy8TCX6UCba4JrUbzq:b-896fix) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/9b9d79cc924df617cf04b27f85957e6da9f58aee/drake_ros/core/test/test_geometry_conversions_py.py#L423)):*
<details><summary><i>Previously, EricCousineau-TRI (Eric Cousineau) wrote…</i></summary><blockquote>

Per #243, we should ensure this has a clause for Bazel to know to execute this code. From #238, it should be something like:
```
if __name__ == '__main__':
    sys.exit(pytest.main(sys.argv))
```

</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

EricCousineau-TRI reviewed (changes requested):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> pending blocking discussion item and CI passing, thanks!

> Reviewed 1 of 3 files at r5, 9 of 9 files at r11.
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-:-NPyDaki4S-NLEPdhzDh:bu1h1q4)* status: 10 of 13 files reviewed, 4 unresolved discussions (waiting on @adityapande-1995 and @sloretz-tri)

*[a discussion](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NPtgjeXA4grqntWOAtG:-NPyDUJA9pJayL38-AbJ:b6b5kd7) (no related file):*
<details><summary><i>Previously, adityapande-1995 (Aditya Pande) wrote…</i></summary><blockquote>

Done, rebased and moved the python test.
</blockquote></details>

Thanks! It looks like there may be one more merge conflict with latest merge.

___
___
*[`drake_ros/drake_ros/core/__init__.py` line 4 at r11](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NPyCXWyDqYzwr1Rj5Pm:-NPyCXWyDqYzwr1Rj5Pn:b-o8ygyz) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/528d1c960418311fc47f19966456fea060c583fe/drake_ros/drake_ros/core/__init__.py#L4)):*
> ```Python
> 
> from drake_ros._cc.core import init
> from drake_ros._cc.core import Isometry3ToRosPose
> ```

BTW I'm concerned about the "scalability" of this level of imports, but won't defect it yet.
___
*[`drake_ros/drake_ros/core/cc_pybind.cc` line 167 at r11](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NPyCz-g9rkQIPFjrEnz:-NPyCz-g9rkQIPFjrEo-:b-e8y80t) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/528d1c960418311fc47f19966456fea060c583fe/drake_ros/drake_ros/core/cc_pybind.cc#L167)):*
> ```cpp
>   // Vector / Translation functions.
>   m.def("RosPointToVector3",
>       &drake_ros_core::RosPointToVector3);
> ```

(For all bound methods) We should bind `py::arg()`, ideally exactly the same as in C++. e.g. for this method, you would add:
```
m.def(..., py::arg("point"));
```
See https://drake.mit.edu/doxygen_cxx/group__python__bindings.html, Conventions > API

In testing, you should then add the kwarg in the invocation, e.g.
```
RosPointToVector3(point=my_point)
```


<!-- Sent from Reviewable.io -->

adityapande-1995 commented:
<details><summary><i>Previously, EricCousineau-TRI (Eric Cousineau) wrote…</i></summary><blockquote>

Thanks! It looks like there may be one more merge conflict with latest merge.
</blockquote></details>

Fixed.




<!-- Sent from Reviewable.io -->

adityapande-1995 commented:
*[`drake_ros/drake_ros/core/cc_pybind.cc` line 167 at r11](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NPyCz-g9rkQIPFjrEnz:-NPyfcre6fgW0YVxTw2z:b-896fix) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/528d1c960418311fc47f19966456fea060c583fe/drake_ros/drake_ros/core/cc_pybind.cc#L167)):*
<details><summary><i>Previously, EricCousineau-TRI (Eric Cousineau) wrote…</i></summary><blockquote>

(For all bound methods) We should bind `py::arg()`, ideally exactly the same as in C++. e.g. for this method, you would add:
```
m.def(..., py::arg("point"));
```
See https://drake.mit.edu/doxygen_cxx/group__python__bindings.html, Conventions > API

In testing, you should then add the kwarg in the invocation, e.g.
```
RosPointToVector3(point=my_point)
```
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

adityapande-1995 commented:
Green CI ! 🎆

EricCousineau-TRI reviewed (commented):
> Reviewed 1 of 3 files at r6, 2 of 3 files at r12, 2 of 3 files at r13, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215)* status: 13 of 14 files reviewed, 2 unresolved discussions (waiting on @adityapande-1995, @sloretz, and @sloretz-tri)

___
*[`drake_ros/core/test/test_geometry_conversions_py.py` line 30 at r5](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NPdmxv_3wKe4twxlstx:-NPys1Cl8WbxGs81v7Q_:b-5oxc8v) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/5fd3f1eebe33c45e112ed401ce6939a51f93fbef/drake_ros/core/test/test_geometry_conversions_py.py#L30)):*
<details><summary><i>Previously, sloretz (Shane Loretz) wrote…</i></summary><blockquote>

nit, this is the converted point. The expected point is `p`.
</blockquote></details>

@adityapande-1995 Was this addressed?
I assume so, looking at `r5..r6`, but want to confirm.
___
*[`drake_ros/core/test/test_geometry_conversions_py.py` line 50 at r5](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NPdnDdy6EMNwSYnW8yy:-NPys3QZ-gTLPhaCQWIc:b-53dj31) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/5fd3f1eebe33c45e112ed401ce6939a51f93fbef/drake_ros/core/test/test_geometry_conversions_py.py#L50)):*
<details><summary><i>Previously, sloretz (Shane Loretz) wrote…</i></summary><blockquote>

nit, this is the converted value, not the expected one.
</blockquote></details>

(same as above)


<!-- Sent from Reviewable.io -->

EricCousineau-TRI reviewed (commented):
sweet! Just two nits left, but I think they're already addressed - just wanna confirm.

> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-:-NPysI12CRLS8u3m1DBF:b-i3z1n6)* status: 13 of 14 files reviewed, 2 unresolved discussions (waiting on @adityapande-1995, @sloretz, and @sloretz-tri)



<!-- Sent from Reviewable.io -->

adityapande-1995 commented:
*[`drake_ros/core/test/test_geometry_conversions_py.py` line 30 at r5](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NPdmxv_3wKe4twxlstx:-NPytkDQEK8ZoL55THMt:bpk4wax) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/5fd3f1eebe33c45e112ed401ce6939a51f93fbef/drake_ros/core/test/test_geometry_conversions_py.py#L30)):*
<details><summary><i>Previously, EricCousineau-TRI (Eric Cousineau) wrote…</i></summary><blockquote>

@adityapande-1995 Was this addressed?
I assume so, looking at `r5..r6`, but want to confirm.
</blockquote></details>

Yup, that was resolved.


<!-- Sent from Reviewable.io -->

adityapande-1995 commented:
*[`drake_ros/core/test/test_geometry_conversions_py.py` line 50 at r5](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-NPdnDdy6EMNwSYnW8yy:-NPytx7c80r8geEeUF2S:b-ellfz4) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/5fd3f1eebe33c45e112ed401ce6939a51f93fbef/drake_ros/core/test/test_geometry_conversions_py.py#L50)):*
<details><summary><i>Previously, EricCousineau-TRI (Eric Cousineau) wrote…</i></summary><blockquote>

(same as above)
</blockquote></details>

Yup, this was resolved. 


<!-- Sent from Reviewable.io -->

EricCousineau-TRI reviewed (approved):
> Reviewed 1 of 3 files at r13.
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @sloretz-tri)



<!-- Sent from Reviewable.io -->

EricCousineau-TRI reviewed (approved):
@sloretz I think this is just waiting for your stamp!

> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake-ros/215#-:-NPyup1QEqfdSL3AwXJn:brxn2b6)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @sloretz-tri)



<!-- Sent from Reviewable.io -->
